api/types/types.go

```diff
@@ -154,8 +154,9 @@ type Version struct {
 	BuildTime     string `json:",omitempty"`
 }
 
-// Commit records a external tool actual commit id version along the
-// one expect by dockerd as set at build time
+// Commit records commits of an external tool, e.g containerd, runC.
+// ID represents the actual commit ID of external tool.
+// Expected represents the commit ID of external tool expected by dockerd as set at build time.
```

Comment:
Still not sure if it's clear with this description (what are "commits of an external tool"?). Just some quick suggestion, but not sure if it's any good;

```Markdown
Commit holds the Git-commit (SHA1), as reported in the version-string
of external tools, such as containerd, or runC.
```

Also; should `ID` and `Expected` be documented in this comment, or separately (as a comment _inside_ the struct?